File: backend/api/routes/pharmacy.py
from flask import Blueprint, jsonify
from services.data_service import data_service

pharmacy_bp = Blueprint('pharmacy', __name__)

@pharmacy_bp.route('/api/pharmacies')
def get_pharmacies():
    """Get pharmacy information with acquisition status"""
    pharmacies = data_service.get_pharmacies()
    
    if not pharmacies:
        return jsonify({'error': 'No data uploaded'}), 400
    
    return jsonify(pharmacies)

@pharmacy_bp.route('/api/clusters')
def get_clusters():
    """Get cluster information with associated pharmacies"""
    clusters = data_service.get_clusters()
    
    if not clusters:
        return jsonify({'error': 'No data uploaded'}), 400
    
    return jsonify(clusters) 

File: backend-js/routes/pharmacy.js
const express = require('express');
const router = express.Router();
const dataService = require('../services/dataService');

router.get('/pharmacies', (req, res) => {
  try {
    const pharmacies = dataService.getPharmacies();
    res.json(pharmacies);
  } catch (error) {
    res.status(500).json({ error: 'Error fetching pharmacies' });
  }
});

router.get('/clusters', (req, res) => {
  try {
    const clusters = dataService.getClusters();
    res.json(clusters);
  } catch (error) {
    res.status(500).json({ error: 'Error fetching clusters' });
  }
});

module.exports = router; 